cpp/include/cudf/utilities/span.hpp

```diff
@@ -258,6 +258,16 @@ class base_2dspan {
     return {this->data() + flatten_index(row, 0, this->size()), this->size().second};
   }
 
+  constexpr base_2dspan subspan(size_t offset, size_t count) const noexcept
```

Comment:
it is named after the corresponding method of 1d span, which itself keeps the naming from [STL](https://en.cppreference.com/w/cpp/container/span/subspan)

Reply:
If we want to follow that convention we can keep the names but have offset/count be the same type as base_2dspan.size(). We can error out for now if unsupported values are used (column offset, column count < size).

Reply:
So no `constexpr` then?

Reply:
Alternatively you can add stride support and enable general subspans. Or we can keep the implementation and rename to something like `rows`, since the version with `size_t `params is easier to use when only row selection is needed.

Reply:
Now that i re-read this https://github.com/rapidsai/cudf/blob/4a02435e08eec61d555943fe687420bd3df1b748/cpp/include/cudf/utilities/span.hpp#L230
I think it's easier to rename the args so that only `size_t` is accepted.

My `constexpr` comment was to point out that i still might not be able to add checking to the args. I assumed it was ok because there's many methods in span/2dspan that could use checking but didn't, in order to keep it `constexpr`.